pkg/cmd/completion/completion.go

```diff
@@ -50,11 +50,13 @@ func Command() *cobra.Command {
 			"commandType": "utility",
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			switch args[0] {
-			case "bash":
-				return cmd.Root().GenBashCompletion(os.Stdout)
-			case "zsh":
-				return runCompletionZsh(os.Stdout, cmd.Parent())
+			if len(os.Args) == 3 {
```

Comment:
Why `== 3` ? :thinking: we are only using `args[0]` so I would guess `== 1` would be the right value :angel:

Reply:
I tested and the `os.Args` comes with value of `2` when executing `tkn completion` and `3` when `tkn completion bash`. I think `os.Arg[0]` is the name of the program, `os.Args[1]` is the `completion` command and then `os.Args[2]` is `bash` or `zsh`.

Reply:
I just noticed that I tested the `os` level object instead of the `args` from Cobra library. The code gets cleaner with `args == 1` instead of `os.Args == 3`. I'll fix that!

Reply:
@davivcgarcia oh indeed, you were using `os.Args` :sweat_smile:, it's better to use `args` from Cobra :wink:

Reply:
Done! :smile: